adcc/gradients/__init__.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python3
+## vi: tabstop=4 shiftwidth=4 softtabstop=4 expandtab
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2021 by the adcc authors
+##
+## This file is part of adcc.
+##
+## adcc is free software: you can redistribute it and/or modify
+## it under the terms of the GNU General Public License as published
+## by the Free Software Foundation, either version 3 of the License, or
+## (at your option) any later version.
+##
+## adcc is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with adcc. If not, see <http://www.gnu.org/licenses/>.
+##
+## ---------------------------------------------------------------------
+from dataclasses import dataclass
+from typing import Dict, Union, Optional
+import numpy as np
+
+from adcc.LazyMp import LazyMp
+from adcc.Excitation import Excitation
+from adcc.timings import Timer
+from adcc.functions import einsum, evaluate
+
+from adcc.OneParticleOperator import OneParticleOperator, product_trace
+from .TwoParticleDensityMatrix import TwoParticleDensityMatrix
+from .orbital_response import (
+    orbital_response, orbital_response_rhs, energy_weighted_density_matrix
+)
+from .amplitude_response import amplitude_relaxed_densities
+
+
+@dataclass(frozen=True)
+class GradientComponents:
+    natoms: int
+    nuc: np.ndarray
```

Comment:
Nuc is a bit short, maybe add a comment here to explain

Reply:
Good idea, I think it would be good to make all attribute names a bit more verbose.